liquidator/index.js

```diff
@@ -189,69 +195,73 @@ async function run(
+      // Default to 3 re-tries on error within the execution loop.
+      errorRetries: process.env.ERROR_RETRIES ? Number(process.env.ERROR_RETRIES) : 5,
+      // Default to 10 seconds in between error re-tries.
+      errorRetriesTimeout: process.env.ERROR_RETRIES__TIMEOUT ? Number(process.env.ERROR_RETRIES__TIMEOUT) : 10000,
+      // Read price feed configuration from an environment variable. This can be a crypto watch, medianizer or uniswap
+      // price feed Config defines the exchanges to use. If not provided then the bot will try and infer a price feed
+      // from the EMP_ADDRESS. EG with medianizer: {"type":"medianizer","pair":"ethbtc",
+      // "lookback":7200, "minTimeBetweenUpdates":60,"medianizedFeeds":[{"type":"cryptowatch","exchange":"coinbase-pro"},
+      // {"type":"cryptowatch","exchange":"binance"}]}
+      priceFeedConfig: process.env.PRICE_FEED_CONFIG ? JSON.parse(process.env.PRICE_FEED_CONFIG) : null,
+      // If there is a liquidator config, add it. Else, set to null. This config contains crThreshold,liquidationDeadline,
+      // liquidationMinPrice, txnGasLimit & logOverrides. Example config:
+      // {"crThreshold":0.02,  -> Liquidate if a positions collateral falls more than this % below the min CR requirement
+      //   "liquidationDeadline":300, -> Aborts if the transaction is mined this amount of time after the last update
+      //   "liquidationMinPrice":0, -> Aborts if the amount of collateral in the position per token is below this ratio
+      //   "txnGasLimit":9000000 -> Gas limit to set for sending on-chain transactions.
+      //   "logOverrides":{"positionLiquidated":"warn"}} -> override specific events log levels.
+      liquidatorConfig: process.env.LIQUIDATOR_CONFIG ? JSON.parse(process.env.LIQUIDATOR_CONFIG) : null,
+      // If there is a LIQUIDATOR_OVERRIDE_PRICE environment variable then the liquidator will disregard the price from the
+      // price feed and preform liquidations at this override price. Use with caution as wrong input could cause invalid liquidations.
+      liquidatorOverridePrice: process.env.LIQUIDATOR_OVERRIDE_PRICE
+    };
 
-    await run(
-      Logger,
-      process.env.EMP_ADDRESS,
-      pollingDelay,
-      errorRetries,
-      errorRetriesTimeout,
-      priceFeedConfig,
-      liquidatorConfig,
-      liquidatorOverridePrice
-    );
+    // Check if the bot is being run as a node process or as a truffle process.
+    if (typeof web3 == "undefined") {
+      // Create a web3 instance. This has built in re-try on error and loads in a provided mnemonic or private key.
+      const { web3 } = require("@umaprotocol/financial-templates-lib/clients/Web3WebsocketClient");
+      await run(Logger, web3, ...Object.values(executionParameters));
+      // Else, if the web3 instance is not undefined, then the script is being run from Truffle. Use present web3 instance.
+    } else {
+      await run(Logger, web3, ...Object.values(executionParameters));
```

Comment:
nit: I think you're implicitly depending on object enumeration order. This means that the order that you specify the keys in the declaration of the `executionParameters` object defines how they'll be passed to this function and interpreted in the function. This isn't super clear, and might be a little confusing if someone were to change any of this code. I would suggest doing either:
1. Changing this function to take an object with named params so it would depend on names rather than ordering:
```js
async function run({ logger, web3, empAddress, ... }) {
...
}
```

Note: I think this change would have a fairly small footprint. I think this syntax would be nice for other bot functions that have many arguments, too.

2. Write a comment explaining the significance of the ordering above the declaration of `executionParameters`.

Reply:
Ye, good point on it not being super clear. It was just so nasty having to call this function in two different contexts, depending on if there was (or was not) a web3 instance. I've opted for change 2, adding a comment explaining the significance of ordering.

I think more broadly change 1 would be great to do in the bots but it should be applied consistently. I don't want to make this change here and only here.

Reply:
SGTM.